include/cuda/std/detail/libcxx/include/experimental/__p0009_bits/trait_backports.hpp

```diff
@@ -0,0 +1,160 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 2.0
+//              Copyright (2019) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef _LIBCUDACXX_EXPERIMENTAL___P0009_BITS_TRAIT_BACKPORTS_HPP
+#define _LIBCUDACXX_EXPERIMENTAL___P0009_BITS_TRAIT_BACKPORTS_HPP
+
+#include "macros.hpp"
+#include "config.hpp"
+
+#include <type_traits>
```

Comment:
Similarly, wrap with `#ifndef __cuda_std__` and add includes for these in `libcudacxx/include/cuda/std/experimental/mdspan`

Reply:
Is this because different `type_traits` gets included when included from here vs from `include/cuda/std/experimental/mdspan`?
Like one picks up `include/cuda/std/detail/libcxx/include/type_traits` and the other picks up `include/cuda/std/type_traits?

Reply:
From what I see in other files, I think I should
- In `libcudacxx/include/cuda/std/experimental/mdspan`, have `#include "type_traits"`
- In files under `__p0009_bits`, have `#include <type_traits>` when `#ifndef __cuda_std__`.